internal/test/wsu/wsu_test.go

```diff
@@ -164,53 +164,73 @@ func testBuiltWMCB(t *testing.T, ansibleOutput string) {
 func TestWSU(t *testing.T) {
 	require.NotEmptyf(t, playbookPath, "WSU_PATH environment variable not set")
 	require.NotEmptyf(t, clusterAddress, "CLUSTER_ADDR environment variable not set")
-
 	require.GreaterOrEqual(t, vmCount, 1, "Expected one or more VMs")
 
 	wmcbPinnedURL, err := getLatestReleasedArtifactURL("wmcb.exe")
 	require.NoError(t, err, "Could not get WMCB url")
-	pinnedWMCBOption := "wmcb_url=" + wmcbPinnedURL
 
 	// Run VM specific tests
 	for i, vm := range framework.WinVMs {
-		func(vmNum int, vm e2ef.WindowsVM) {
-			t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
-				// Indicate that we can run the test suite on each node in parallel
-				t.Parallel()
-
-				// In order to run the ansible playbook we create an inventory file:
-				// https://docs.ansible.com/ansible/latest/user_guide/intro_inventory.html
-				// Passing in a slice with a single entry for now
-				hostFilePath, err := createHostFile([]e2ef.WindowsVM{framework.WinVMs[vmNum]})
-				require.NoErrorf(t, err, "Could not write to host file: %s", err)
-
-				// Run the playbook
-				// For the first VM we use a pinned WMCB version
-				var ansibleCmd *exec.Cmd
-				var pinnedWMCB bool
-				if vmNum == 0 {
-					pinnedWMCB = true
-					ansibleCmd = exec.Command("ansible-playbook", "-v", "-e", pinnedWMCBOption, "-i",
-						hostFilePath, playbookPath)
-				} else {
-					ansibleCmd = exec.Command("ansible-playbook", "-v", "-i", hostFilePath, playbookPath)
-				}
-				out, err := ansibleCmd.CombinedOutput()
-				require.NoError(t, err, "WSU playbook returned error: %s, with output: %s", err, string(out))
-				runE2ETestSuite(t, pinnedWMCB, vm, string(out))
-
-				// Run the test suite twice, to ensure that the WSU can be run multiple times against the same VM
-				t.Run("Run the WSU against the same VM again", func(t *testing.T) {
-					runE2ETestSuite(t, pinnedWMCB, vm, string(out))
-				})
-			})
-		}(i, vm)
+		t.Run("VM "+strconv.Itoa(i), func(t *testing.T) {
+			testVM(t, i, vm, wmcbPinnedURL)
+		})
 	}
 
 	// Run cluster wide tests
 	t.Run("Pending CSRs were approved", testNoPendingCSRs)
 }
 
+// testVM runs the WSU against the given VM and runs the e2e test suite against that VM as well
+func testVM(t *testing.T, vmNum int, vm e2ef.WindowsVM, wmcbURL string) {
+	var wsuOut string
+	var pinnedWMCB bool
+	var err error
+	pinnedWMCBOption := "wmcb_url=" + wmcbURL
```

Comment:
nit: Having a slice to pass around and convert to string later might be helpful to keep track for all the vars.

Reply:
@ravisantoshgudimetla I don't understand your nit. Please add a code snippet explaining what you want done.

Reply:
This is similar to what we did last time in wmcb, we'll create a 
``` go
args := []string{"wmcb_url=abc", "arg2=val2"} 
```

have a function like:
```
go
func generateAnsibleArgsString(args []string) string {
argString = ""
for _, arg := range args {
    argString += arg
}
return argString
}
```

This ensures we can validate the inputs we want in future. 
PS: Better would be a map but since we have one arg as of now, so I am fine.